uclasm/matching/local_cost_bound/edgewise.py

```diff
@@ -533,44 +533,30 @@ def add_time_costs(smp, candidates, local_costs):
             cand1_sorted_idxs = cand1_times.argsort()
             cand2_sorted_idxs = cand2_times.argsort()
 
-            cand2_idx_idx = 0
-            cand2_idx = cand2_sorted_idxs[cand2_idx_idx]
-            cand2_time = cand2_times[cand2_idx]
-            last_valid_cand2_idx_idx = 0
-            for cand1_idx_idx, cand1_idx in enumerate(cand1_sorted_idxs):
-                cand1_time = cand1_times[cand1_idx]
-                while cand2_time - cand1_time < min_timedelta:
-                    cand2_idx_idx += 1
-                    if cand2_idx_idx >= len(cand2_sorted_idxs):
-                        cand2_time = None
-                        break
-                    cand2_idx = cand2_sorted_idxs[cand2_idx_idx]
-                    cand2_time = cand2_times[cand2_idx]
-                if cand2_time is not None:
-                    if 'maxValue' in time_constraint:
-                        if cand2_time - cand1_time > max_timedelta:
-                            continue
-                    cand1_nonnat_costs[cand1_idx] = 0
-                    if 'maxValue' in time_constraint:
-                        cand2_nonnat_costs[cand2_idx] = 0
-                        temp_cand2_idx_idx = max(last_valid_cand2_idx_idx, cand2_idx_idx) + 1
-                        if temp_cand2_idx_idx < len(cand2_sorted_idxs):
-                            next_cand2_idx = cand2_sorted_idxs[temp_cand2_idx_idx]
-                            next_cand2_time = cand2_times[next_cand2_idx]
-                            while next_cand2_time - cand1_time <= max_timedelta:
-                                cand2_nonnat_costs[next_cand2_idx] = 0
-                                last_valid_cand2_idx_idx = temp_cand2_idx_idx
-                                temp_cand2_idx_idx += 1
-                                if temp_cand2_idx_idx < len(cand2_sorted_idxs):
-                                    next_cand2_idx = cand2_sorted_idxs[temp_cand2_idx_idx]
-                                    next_cand2_time = cand2_times[next_cand2_idx]
-                                else:
-                                    break
-                    elif cand1_idx_idx == 0:
-                        # With only minimum timedelta, all future cand2_times are valid
-                        cand2_nonnat_costs[cand2_sorted_idxs[cand2_idx_idx:]] = 0
+            # There are a lot of redundant times, only check unique times
+            uniq_cand1_times = np.unique(cand1_times)
+
+            # For each initial event time
+            for cand1_time in uniq_cand1_times:
+                if 'minValue' in time_constraint:
+                    min_idx = np.searchsorted(cand2_times, cand1_time + min_timedelta,
+                                              sorter=cand2_sorted_idxs)
                 else:
-                    break
+                    min_idx = 0
```

Comment:
If no minValue is set, it should be a min_timedelta of 0, since the two events always have to happen in order. This already gets taken care of earlier when min_timedelta is set, so this case can get ignored.

Reply:
Fixed.